R/get_coding_ssm_status.R

```diff
@@ -41,49 +80,108 @@
+    these_samples_metadata,
+    from_flatfile = TRUE,
+    augmented = TRUE,
+    min_read_support = 3,
+    maf_path = NULL,
+    maf_data,
+    include_hotspots = TRUE,
+    keep_multihit_hotspot = FALSE,
+    recurrence_min = 5,
+    this_seq_type = "genome",
+    projection = "grch37",
+    review_hotspots = TRUE,
+    genes_of_interest = c("FOXO1", "MYD88", "CREBBP"),
+    genome_build = "hg19",
+    include_silent = FALSE,
+    include_silent_genes
+){
+  
+  message(
+    "Using the GAMBLR.results version of this function."
+  )
 
   if(missing(gene_symbols)){
     message("defaulting to all lymphoma genes")
     gene_symbols = pull(GAMBLR.data::lymphoma_genes, Gene)
   }
 
+  if(!missing(include_silent_genes)){
+    message(
+        "Combining genes specified with gene_symbols and include_silent_genes"
+    )
+    gene_symbols <- c(
+        gene_symbols,
+        include_silent_genes
+    ) %>%
+    unique()
+  }
+
   if(missing(these_samples_metadata)){
     these_samples_metadata = get_gambl_metadata()
   }
 
+
+  coding_class <- c(
+    "Frame_Shift_Del", "Frame_Shift_Ins", "In_Frame_Del",
+    "In_Frame_Ins", "Missense_Mutation", "Nonsense_Mutation",
+    "Nonstop_Mutation", "Splice_Region", "Splice_Site",
+    "Targeted_Region", "Translation_Start_Site"
+  )
+
+  if(include_silent){
+    message("Including Silent variants")
+    coding_class <- c(coding_class, "Silent")
+  }
+
   # call it once so the object can be reused if user wants to annotate hotspots
   if(!missing(maf_data)){
-    coding_ssm = maf_data %>%
-      dplyr::filter(Variant_Classification %in% coding_class)
+    coding_ssm = maf_data
 
   }else if (!is.null(maf_path)){
     coding_ssm = fread_maf(maf_path)
-    coding_ssm = coding_ssm %>%
-      dplyr::filter(Variant_Classification %in% coding_class)
   }
 
   if(missing(maf_data) & is.null(maf_path)){
-    coding_ssm = get_coding_ssm(projection = projection, this_seq_type = this_seq_type, from_flatfile = from_flatfile, augmented = augmented, min_read_support = 3, basic_columns = FALSE, include_silent = include_silent)
+    coding_ssm = get_coding_ssm(
+        projection = projection,
+        this_seq_type = this_seq_type,
+        from_flatfile = from_flatfile,
+        augmented = augmented,
+        min_read_support = 3,
+        basic_columns = FALSE,
+        include_silent = include_silent
+    )
   }
 
-  coding_ssm = coding_ssm %>%
-    dplyr::filter(Variant_Classification %in% coding_class)
+  if(missing(include_silent_genes)){
+    coding_ssm <- coding_ssm %>%
+        dplyr::filter(
+            Variant_Classification %in% coding_class
+        )
+  } else {
+    message(
+        strwrap(
+            prefix = " ",
+            initial = "", 
+            "You have provided gene list with argument include_silent_genes.
+            The Silent variants will be included even if the include_silent
+            argument is set to FALSE.
+            "
+        )
+    )
+    coding_ssm <- coding_ssm %>%
```

Comment:
Just realized something I hadn't considered. What if we allowed ALL variants in the gene instead of just Silent? For things like LymphGen, it's usually that the person would want to have all mutation classes count, not just Silent ones, right?